### PR TITLE
Convert to sidecar charm

### PR DESCRIPTION
**Pull request by knkski**

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -143,67 +143,41 @@ def main(self, event):
...
+
+    def set_manifest(self, manifest):
+        client = Client()
+        errors = []
+
+        for resource in manifest:
+            try:
+                client.create(resource)
+            except ApiError as err:
+                if err.status == 409:
```
Comment (ca-scribner) [inline src/charm.py:175]: is this tested?  I **think** it should be:
```suggestion
                if err.status.code == 409:
```
but I might misunderstand something

Comment (knkski) [reply src/charm.py:175]: Nope, not tested, will push up update that fixes that and a few other things

**Review by DomFleischmann — COMMENTED** [review]
Although I think this is interesting work I don't feel comfortable with merging this so close to a release. I would prefer to stick with the pod_spec charm for now. We are on a tight deadline and there isn't the time to flesh out all the ideas and changes that are in this PR. Let's stick with handling the status with our pod spec charms correctly for this release. I don't want unpleasant surprises because we rushed a sidecar rewrite to meet a deadline.

**Review by DomFleischmann — APPROVED** [review]
LGTM

**Pull request merged**